packages/client/src/schedule-types.ts

```diff
@@ -0,0 +1,830 @@
+import { checkExtends, Replace, RequireAtLeastOne } from '@temporalio/common/lib/type-helpers';
+import { SearchAttributes, Workflow } from '@temporalio/common';
+import type { temporal } from '@temporalio/proto';
+import { WorkflowStartOptions } from './workflow-client';
+
+export interface UpdateScheduleOptions {
+  force: boolean;
+}
+
+export interface ListScheduleEntry {
+  /**
+   * Schedule Id
+   *
+   * We recommend using a meaningful business identifier.
+   */
+  scheduleId: string;
+
+  /** When Actions are taken */
+  spec: RequireAtLeastOne<ScheduleSpecDescription, 'calendars' | 'intervals'>;
+
+  /**
+   * Additional non-indexed information attached to the Schedule. The values can be anything that is
+   * serializable by the {@link DataConverter}.
+   */
+  memo?: Record<string, any>;
+
+  /**
+   * Additional indexed information attached to the Schedule. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom Data Converter is provided.
+   */
+  searchAttributes: SearchAttributes;
+
+  /**
+   * Informative human-readable message with contextual notes, e.g. the reason
+   * a Schedule is paused. The system may overwrite this message on certain
+   * conditions, e.g. when pause-on-failure happens.
+   */
+  note?: string;
+
+  /**
+   * Whether Schedule is currently paused.
+   *
+   * @default false
+   */
+  paused: boolean;
+
+  /**
+   * Present if action is a {@link StartWorkflowAction}.
+   */
+  workflowType?: string;
```

Comment:
I'm fine with not having the full action, I don't think it's needed.
I would want to structure this differently in TS:

```ts
interface WorkflowScheduleListAction {
  type: 'workflow';
  workflowType: string;
}

// In the future this will be a union type with other actions
type ScheduleListAction = WorkflowScheduleListAction;

// This has to map to something and can be switched on
action: ScheduleListAction;
```